cmd/runner/taskrunner/taskrunner.go

```diff
@@ -83,18 +83,18 @@ func New(cfg *Config) (*TaskRunner, error) {
 }
 
 func (tr *TaskRunner) ParsePortfolio(ctx context.Context, req *task.ParsePortfolioRequest) (task.ID, task.RunnerID, error) {
-	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(req.AssetIDs); err != nil {
-		return "", "", fmt.Errorf("failed to encode asset IDs: %w", err)
+	var taskBuffer bytes.Buffer
+	if err := json.NewEncoder(&taskBuffer).Encode(req); err != nil {
+		return "", "", fmt.Errorf("failed to encode ParsePortfolioRequest: %w", err)
 	}
 	return tr.run(ctx, []task.EnvVar{
 		{
 			Key:   "TASK_TYPE",
 			Value: string(task.ParsePortfolio),
 		},
 		{
-			Key:   "ASSET_IDS",
-			Value: buf.String(),
+			Key:   "PARSE_PORTFOLIO_REQUEST",
+			Value: taskBuffer.String(),
```

Comment:
It's fine to dump everything in a single env var as long as all the contents are bounded. The second we have arrays of things, we're headed for trouble, as [there's a 128KB limit on env var size](https://askubuntu.com/a/1385554).

Granted, this was true before, but for anything _really_ large, I'd serialize it to blob store and send over the URI

Reply:
Agreed generally. I've now added in a harder failure mode to alert us if/when this hits that limit.